runtime/compiler/runtime/asmprotos.h

```diff
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 
-#if (defined(__IBMCPP__) || defined(__IBMC__) && !defined(MVS)) && !defined(J9ZOS390) && !defined(LINUXPPC64)
+#if (defined(__IBMCPP__) || defined(__IBMC__) && !defined(MVS)) && !defined(J9ZOS390) && !defined(LINUXPPC64) || defined(__open_xl__)
```

Comment:
this looks weird, given the fact that && has higher precedence over ||.  it seems dangerous or unreadable to mix up compiler macro (__open_xl__) with platform macro (J9ZOS390 and LINUXPPC64).

Reply:
@r30shah this looks an existing problem as well re MVS.  i think MVS is also a platform macro (i.e. not defined by the compiler. but defined to indicate the particular platform). and, I think it should be `__MVS__` to begin with.  if @r30shah agrees, we should fix this as well.

Reply:
I agree, it should be `__MVS__` and with the fix Julian suggests.

Reply:
As the MVS issue is not introduced by the changes of this PR I will address this in future creating a new PR.